frontend/apps/app/app/(app)/app/organizations/[organizationId]/settings/general/GeneralPage.tsx

```diff
@@ -1,129 +1,136 @@
+  const organization = await getOrganizationDetails(organizationId)
+  if (!organization) {
+    redirect(
+      `/app/organizations/${organizationId}/settings/general?error=${encodeURIComponent('Organization not found')}`,
+    )
+  }
+
+  // Verify that the confirmation text matches the organization name
+  const confirmText = formData.get('confirmText')
+  if (confirmText !== organization.name) {
+    redirect(
+      `/app/organizations/${organizationId}/settings/general?error=${encodeURIComponent('Confirmation text does not match organization name')}`,
+    )
   }
 
-  const handleSave = async () => {
-    if (!organizationData.name.trim()) {
-      setOrganizationData((prev) => ({
-        ...prev,
-        error: 'Organization name cannot be empty',
-      }))
-      return
-    }
-
-    if (!organizationId) {
-      setOrganizationData((prev) => ({
-        ...prev,
-        error: 'Organization ID is missing',
-      }))
-      return
-    }
-
-    setOrganizationData((prev) => ({
-      ...prev,
-      isLoading: true,
-      error: null,
-      success: false,
-    }))
-
-    const supabase = createClient()
-    const { error } = await supabase
-      .from('organizations')
-      .update({ name: organizationData.name.trim() })
-      .eq('id', organizationId)
-
-    if (error) {
-      setOrganizationData((prev) => ({
-        ...prev,
-        error: 'Failed to update organization name',
-        isLoading: false,
-      }))
-      console.error(error)
-      return
-    }
-
-    setOrganizationData((prev) => ({
-      ...prev,
-      success: true,
-      isLoading: false,
-    }))
-    toast({
-      title: 'Updated',
-      description: 'Organization name has been successfully updated',
-      status: 'success',
-    })
-    setTimeout(() => {
-      setOrganizationData((prev) => ({ ...prev, success: false }))
-    }, 3000) // Clear success message after 3 seconds
+  const result = await deleteOrganization(organizationId)
+
+  if (!result.success) {
+    redirect(
+      `/app/organizations/${organizationId}/settings/general?error=${encodeURIComponent(result.error || 'Failed to delete organization')}`,
+    )
   }
 
-  const handleDelete = () => {
-    // TODO: Implement delete functionality
-    // Delete organization
+  // Redirect with success message and additional parameters for toast notification
+  redirect(
+    `/app/organizations?success=Organization "${organization.name}" has been deleted successfully`,
+  )
+}
+
+export async function GeneralPage({
+  organizationId,
+  searchParams,
+}: {
+  organizationId: string
+  searchParams?: { [key: string]: string | string[] | undefined }
+}) {
+  // Fetch organization details using the existing function
+  const organization = await getOrganizationDetails(organizationId)
+
+  if (!organization) {
+    // Handle case where organization is not found
+    return (
+      <div className={styles.container}>
+        <p>Organization not found</p>
+      </div>
+    )
```

Comment:
It is better to return a 404 status and error page here.

```suggestion
    // Handle case where organization is not found
    return notFound()
```

Reply:
Thank you🙏 I fixed it.
https://github.com/liam-hq/liam/pull/1420/commits/0efd6089cdf7d57c5b531c4ff44f70c48ee86a87